src/hooks/useSwapInputHandlers.ts

```diff
@@ -28,16 +35,45 @@ export default function useSwapInputHandlers() {
     if (type === ExchangeModalTypes.withdrawal) {
       dispatch(updateSwapInputAmount(supplyBalanceUnderlying));
     } else {
-      let amount =
-        ethereumUtils.getAccountAsset(inputCurrencyAddress)?.balance?.amount ??
-        '0';
-      if (inputCurrencyAddress === ETH_ADDRESS) {
-        const remaining = subtract(amount, MIN_ETH);
-        amount = greaterThan(remaining, 0) ? remaining : '0';
+      const accountAsset = ethereumUtils.getAccountAsset(inputCurrencyAddress);
+      const oldAmount = accountAsset?.balance?.amount ?? '0';
+      let newAmount = oldAmount;
+      if (inputCurrencyAddress === ETH_ADDRESS && accountAsset) {
+        newAmount = toFixedDecimals(
+          ethereumUtils.getBalanceAmount(selectedGasFee, accountAsset),
+          6
+        );
+
+        Alert({
+          buttons: [
+            { style: 'cancel', text: 'No thanks' },
+            {
+              onPress: () => {
```

Comment:
I'm not a fan of this auto-adjusting. I'd rather just display the warning and then let the user adjust. What if the fee will get 2x higher what is quite realistic?

Reply:
Yeah I kinda agree too – it was suggested on Linear that it could be 1.5x the txn amount left over. @jinchung what do you think? Should we omit auto-adjust?

Reply:
this was part of the spec from @christianbaroni - need to discuss with him if he wants to change it